sogs/migrations/v_0_1_x.py

```diff
@@ -27,6 +27,14 @@ def migrate(conn):
     return True
 
 
+def sqlite_connect_readonly(path):
+    import sqlite3
+
+    conn = sqlite3.connect(f"file:{path}?mode=ro", uri=True, isolation_level=None)
+    conn.row_factory = sqlite3.Row
+    return conn
```

Comment:
since this is used in a `with` statement does returning here work instead of using a `yield`?